packages/ext-taquito/src/tzktReadProvider.ts

```diff
@@ -0,0 +1,195 @@
+      const {data} = await blocksGetCount();
+      return data
+    }
+
+    if(String(block).includes('head~')) {
+      const n = Number(String(block).split('head~')[1])
+      const {data} = await blocksGetCount();
+
+      return  Number(data) - n
+    }
+
+    const {data} = await blocksGetByHash(block);
+    return data?.level || 0
+  }
+
+  async getEntrypoints(contract: string): Promise<EntrypointsResponse> {
+    const {data} = await contractsGetEntrypoints(contract);
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    return data;
+  }
+
+  async getScript(address: string): Promise<ScriptedContracts> {
+    const {data} = await contractsGetStorageSchema(address);
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    return data;
+  }
+
+  async getBalance(address: string, block: BlockIdentifier): Promise<BigNumber> {
+    const blockId = await this._getBlockIdentifier(block);
+    const {data} = await accountsGetBalanceAtLevel(address, blockId);
+    return new BigNumber(data);
+  }
+
+  async getDelegate(address: string): Promise<string | null> {
+    const {data} = await accountsGetByAddress(address);
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    return data?.delegate?.address
+  }
+
+  async getNextProtocol(): Promise<string> {
+    const {data} = await protocolsGetCurrent()
+    return data?.hash || ''
+  }
+
+  async getProtocolConstants(block: BlockIdentifier): Promise<{
+    time_between_blocks?: BigNumber[] | undefined;
+    minimal_block_delay?: BigNumber | undefined;
+    hard_gas_limit_per_operation: BigNumber;
+    hard_gas_limit_per_block: BigNumber;
+    hard_storage_limit_per_operation: BigNumber;
+    cost_per_byte: BigNumber;
+  }> {
+
+    let protocol: Protocol
+
+    if(block) {
+      const blockId = await this._getBlockIdentifier(block)
+      const blockByLevel = await blocksGetByLevel(blockId);
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      const {data} = await protocolsGetByCycle(blockByLevel.data.cycle || 0)
+      protocol = data;
+    } else {
+      const {data} = await protocolsGetCurrent()
+      protocol = data;
+    }
+
+    return {
+      time_between_blocks: [new BigNumber(protocol?.constants?.timeBetweenBlocks || 0)] || undefined,
+      minimal_block_delay: new BigNumber(0),
+      hard_gas_limit_per_operation: new BigNumber(protocol?.constants?.hardOperationGasLimit || 0),
+      hard_gas_limit_per_block: new BigNumber(protocol?.constants?.hardBlockGasLimit || 0),
+      hard_storage_limit_per_operation: new BigNumber(protocol?.constants?.hardOperationStorageLimit || 0),
+      cost_per_byte: new BigNumber(protocol?.constants?.byteCost || 0),
+    }
+  }
+
+  async getStorage(contract: string, block: BlockIdentifier): Promise<MichelsonV1Expression> {
+    const blockId = await this._getBlockIdentifier(block)
```

Comment:
If block is 'head' you don't need to query its level just call get Storage without filters